### PR TITLE
Add concurrency limits

### PR DESCRIPTION
**Pull request by giovannibenussi**
This PR to limits the number of parallel executions when calling the `execute`, `batch`, `transaction`, and `executeMultiple` methods. The number of parallel executions that can be executed is 20 by default, and can be customized by providing a new `concurrency` parameter when initializing the client as follows:

```js
export const turso = createClient({
  url: process.env.TURSO_DATABASE_URL,
  authToken: process.env.TURSO_AUTH_TOKEN,
  concurrency: 10 // defaults to 20 and undefined means there's no concurrency limit
});
```

### CODE REVIEW

**Comment by giovannibenussi** [comment]
For context of why `p-limit` didn't work: it's a pure ESM package, which makes it really hard to work with Jest, even when following the official recommendations from Jest docs on the subject.

File: packages/libsql-client/src/http.ts [hunk 1]
```diff
@@ -99,33 +102,35 @@ export class HttpClient implements Client {
...
     async execute(stmt: InStatement): Promise<ResultSet> {
-        try {
-            const isSchemaDatabasePromise = this.getIsSchemaDatabase();
-            const hranaStmt = stmtToHrana(stmt);
-
-            // Pipeline all operations, so `hrana.HttpClient` can open the stream, execute the statement and
-            // close the stream in a single HTTP request.
-            let rowsPromise: Promise<hrana.RowsResult>;
-            const stream = this.#client.openStream();
+        return limit(async () => {
```
Comment (athoscouto) [inline packages/libsql-client/src/http.ts:105]: This should probably be added to `batch`, `transaction` and `executeMultiple` as well.
We should do the same in the web socket client.

File: packages/libsql-client/src/http.ts [hunk 2]
```diff
@@ -24,9 +24,12 @@ import {
     getIsSchemaDatabase,
     waitForLastMigrationJobToFinish,
 } from "./migrations.js";
+import promiseLimit from "promise-limit";
 
 export * from "@libsql/core/api";
 
+const limit = promiseLimit<ResultSet>(1);
```
Comment (athoscouto) [inline packages/libsql-client/src/http.ts:31]: I don't think we should use a global one. We should go with a limit per client.
It protects us less, but gives us more predictability.

We should allow users to override the limit value.
I suggest starting with something low, like 20, and exposing a field at the client config object allow users to override it - maybe disable it?

Comment (giovannibenussi) [reply packages/libsql-client/src/http.ts]: I just deployed a change that implements this:

```js
export const turso = createClient({
  url: process.env.TURSO_DATABASE_URL,
  authToken: process.env.TURSO_AUTH_TOKEN,
  concurrency: 10 // defaults to 20 and undefined means there's no limit
});
```

This limit is now per client.

File: packages/libsql-client/src/http.ts (hunk 1 above)
Comment (giovannibenussi) [reply packages/libsql-client/src/http.ts]: Added! The only issue is that we lost type safety while adding all these calls to a single queue:

```
this.#limit = promiseLimit<any>(concurrency);
```

This means that if the return type doesn't match, we won't get a TypeScript error, although the code still works.

An alternative would be to keep 4 queues so each function has its own queue (`execute`, `batch`, `transaction`, and `executeMultiple`), but for now we are keeping one.

I filled an issue in the [promise-limit](https://github.com/featurist/promise-limit/issues/23) repo to see if we can get support for types on a per-call basis.

**Review by athoscouto — APPROVED** [review]

**Comment by giovannibenussi** [comment]
I added an update to the behavior of the `concurrency` value:

- `undefined` will be equivalent to `20` (same if the parameter is not provided).
- Negative numbers will be treated as `0`, which means that there's no concurrency limit.
- Any positive number will limit the concurrency as expected.

**Pull request merged**